coclico/metrics/metric.py

```diff
@@ -6,19 +6,76 @@
 
 
 class Metric:
+    """Base class for metrics"""
+
     def __init__(self, store: Store, class_weights: Dict):
+        """Initialize Metric object
+
+        Args:
+            store (Store): Store object (as defined in gpao_utils) to handle mount points of distant stores on various
+            computers
+            class_weights (Dict): Dictionnary of the weights of the metric for each class (cf. config file)
+        """
         self.store = store
         self.class_weights = class_weights
 
     def create_metric_intrinsic_jobs(
-        self, name: str, tile_names: List[str], input_path: Path, out_path: Path
+        self, name: str, tile_names: List[str], input_path: Path, out_path: Path, is_ref: bool
     ) -> List[Job]:
-        return [self.create_metric_intrinsic_one_job(name, input_path / f, out_path) for f in tile_names]
+        """Create jobs for a single input classification result folder (eg. ref or c1 or c2 or ...)
+        These jobs are aimed to compute intermediate results relative to the input las only (without comparison) that
+        will be used in `create_metric_relative_to_ref_jobs` to compute the relative metric
+        Args:
+            name (str): classification name (used for job name creation)
+            tile_names (List[str]): list of the filenames of the tiles on which to calculate the result
+            input_path (Path): input folder path (path to the results of the classification)
+            out_path (Path): path for the intermediate results to be saved
+            is_ref (bool): flag that says if the input classification folder is the reference folder (in case it has to
+            be treated differently)
+
+        Returns:
+            List[Job]: List of GPAO jobs to create
+        """
+        return [self.create_metric_intrinsic_one_job(name, input_path / f, out_path, is_ref) for f in tile_names]
+
+    def create_metric_intrinsic_one_job(self, name: str, input: Path, output: Path, is_ref: bool) -> Job:
+        """Create a job for a single input classification result file (one tile).
+        ie. job to create of one element of the `tiles_names` list in create_metric_intrinsic_jobs
```

Comment:
J'ai un pb avec `classification result` dans le sens où c'est le resultat d'un process hors de ce projet. Je pense que c'est le fait d'avoir en même temps `input` et `result` qui est confusant. Et la seconde partie , ie... , je trouve ça compliqué à comprendre, et ça apporte pas grand chose au développeur d'une nouvelle métrique. Suggestion plus simple :
`Create a job, to compute intrinsect metric, for a single pointcloud file.`

Reply:
le truc c'est que c'était pas classification result la bonne association  il y a d'un côté `input classification` et de l'autre `result file` mais apparemment c'est pas clair alors je vais changer

Reply:
Je vais changer aussi dans les autres métriques d'ailleurs

Reply:
Yep c'est pas très clair, je vais changer